connection.go

```diff
@@ -158,6 +159,13 @@ func (i *idleAwareFramer) WriteFrame(frame spdy.Frame) error {
 		return err
 	}
 
+	if i.ignorePingFrames {
```

Comment:
Accessing this value should still be accessed with a lock. Maybe switching the timeout lock to rw would work here.

Reply:
I'm not an expert but I always was recommended to not use RWMutex unless I have a big number of goroutines reading , some golang issues claim it is not very performant https://github.com/golang/go/issues/17973
In this case we are protecting a variable that is not likely to change so often, so I think that a simple mutex will not make a big difference, unless the user keeps changing the idletimeout very often :)